scripts/irods/controller.py

```diff
@@ -32,6 +32,35 @@ def check_config(self):
         _ = self.config.server_config
         _ = self.config.version
 
+    def migrate_rei_files_into_catalog(self):
+        l = logging.getLogger(__name__)
+
+        rei_file_dir = os.path.join(paths.irods_directory(), 'config', 'packedRei')
+
+        if not os.path.isdir(rei_file_dir) or len(os.listdir(rei_file_dir)) == 0:
+            return
+
+        db_config = self.config.server_config['plugin_configuration']['database'].values()[0]
+        db_name = 'iRODS Catalog'
+        db_user = db_config['db_username']
+        db_password = db_config['db_password']
+
+        l.info('Migrating REI files to catalog ...')
+        cmd = [paths.migrate_rei_files_into_catalog_executable(), rei_file_dir, db_name, db_user, db_password]
```

Comment:
Does this leak the password into the process status (i.e another user could see the password via `ps`)?

Reply:
Maybe.  How do we want to approach this?

It is implemented this way for flexibility and simplicity.

Do we lose anything by having the migration app read the server config and pull that db info?
Can we guarantee that the REI file directory is always the same path?

Reply:
all fine questions.   will need to think through non-package installs perhaps.   but i think reading the database password from the config *should* be safe... can discuss.